src/shared/regions/defaultRegionProvider.ts

```diff
@@ -29,14 +27,20 @@ export class DefaultRegionProvider implements RegionProvider {
         return this.onRegionProviderUpdatedEmitter.event
     }
 
-    public async getRegionData(): Promise<RegionInfo[]> {
-        return this._loadedRegions
-    }
-
     public isServiceInRegion(serviceId: string, regionId: string): boolean {
         return !!this.regionIdToRegionData.get(regionId)?.serviceIds.find(x => x === serviceId) ?? false
     }
 
+    public getPartitionId(regionId: string): string | undefined {
+        return this.regionIdToRegionData.get(regionId)?.partitionId ?? undefined
+    }
+
+    public getRegions(partitionId: string): Region[] {
```

Comment:
Nit: make this a bit more specific (`getRegionsForPartition`)

Reply:
Can you think of alternative region retrieval for the `RegionProvider` interface that would collide with `getRegions`?

Reply:
Is there a possibility we'd ever want to list all available regions? Can't think of a specific reason why but if I saw this API I'd probably think of it as a "get all regions that this region provider can provide".

It's a nit though, feel free to disregard.